ui/src/containers/NodePageMetricsTab.js

```diff
@@ -212,8 +236,65 @@ const NodePageMetricsTab = (props) => {
   };
   const lineConfig = { strokeWidth: 1.5 };
 
+  // write the selected timespan in URL
+  const writeUrlTimeSpan = (timespan) => {
+    let formatted = queryTimeSpansCodes.find((item) => item.value === timespan);
+
+    if (formatted) {
+      query.set('from', formatted.label);
+      history.push({ search: query.toString() });
+    }
+  };
+
+  const metricsTimeSpanItems = [
+    {
+      label: LAST_SEVEN_DAYS,
+      onClick: () => {
+        dispatch(
+          updateNodeStatsAction({ metricsTimeSpan: LAST_SEVEN_DAYS }),
+          updateNodeStatsFetchArgumentAction(),
+        );
+        writeUrlTimeSpan(LAST_SEVEN_DAYS);
+      },
+      selected: metricsTimeSpan === LAST_SEVEN_DAYS,
+    },
+    {
+      label: LAST_TWENTY_FOUR_HOURS,
+      onClick: () => {
+        dispatch(
+          updateNodeStatsAction({ metricsTimeSpan: LAST_TWENTY_FOUR_HOURS }),
+          updateNodeStatsFetchArgumentAction(),
+        );
+        writeUrlTimeSpan(LAST_TWENTY_FOUR_HOURS);
+      },
+      selected: metricsTimeSpan === LAST_TWENTY_FOUR_HOURS,
+    },
+    {
+      label: LAST_ONE_HOUR,
+      onClick: () => {
+        dispatch(
+          updateNodeStatsAction({ metricsTimeSpan: LAST_ONE_HOUR }),
+          updateNodeStatsFetchArgumentAction(),
+        );
+        writeUrlTimeSpan(LAST_ONE_HOUR);
+      },
+      selected: metricsTimeSpan === LAST_ONE_HOUR,
+    },
+  ];
```

Comment:
Could be refactored:

```suggestion
  const metricsTimeSpanItems = [
    LAST_SEVEN_DAYS, LAST_TWENTY_FOUR_HOURS, LAST_ONE_HOUR
  ].map(
    option => ({
      label: option,
      onClick: () => {
        dispatch(
          updateNodeStatsFetchArgumentAction({ metricsTimeSpan: option }),
        );
        writeUrlTimeSpan(option);
      },
      selected: metricsTimeSpan === option,
    })
  );
```